content/2024-04-11-pq-openmls.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "Post-Quantum OpenMLS"
+date: 2024-04-11
+tags: ["mls", "pq", "openmls"]
+description: "Get started with post-quantum secure end-to-end encrypted with OpenMLS."
+notoc: true
+author: "Franziskus Kiefer"
+author_link: "https://franziskuskiefer.de"
+---
+
+OpenMLS now offers security against harvest-now-decrypt-later (HNDL) quantum adversaries.
+
+In [#1546](https://github.com/openmls/openmls/pull/1546) we merged support for the [X-Wing KEM draft](https://www.ietf.org/archive/id/draft-connolly-cfrg-xwing-kem-02.html), which is an early draft for securely combining elliptic-curve-based Diffie-Hellman with ML-KEM. In particular, OpenMLS now supports the ciphersuite MLS_256_XWING_CHACHA20POLY1305_SHA256_Ed25519 with ciphersuite 0x004D. There is no IANA code-point for this ciphersuite yet, such that interoperability may not be guaranteed. We work with other implementers towards interoperability of this ciphersuite.
+
+The implementation uses Cryspen’s [formally verified ML-KEM](https://cryspen.com/post/ml-kem-verification/) and x25519 implementations from [libcrux](https://github.com/cryspen/libcrux/). The implementations are not only formally verified for correctness, secret independence, and memory safety, but also amongst the fastest implementations. Users should not notice any significant performance differences when using this new ciphersuite.
+The threat of HNDL attackers requires applications to switch to post-quantum secure mechanisms now, just like [Signal](https://signal.org/blog/pqxdh/) and [iMessage](https://security.apple.com/blog/imessage-pq3/) did already. OpenMLS offers a simple way to achieve security against HNDL attackers and is ready to use.
+
+## Performance
+
+While the new ML-KEM mechanism is very efficient, it requires larger messages and because the ciphersuite used is hybrid, the workload increases.
```

Comment:
```suggestion
While the new ML-KEM mechanism is very efficient, it requires larger messages and because the ciphersuite used is hybrid (i.e. a combination of ML-KEM the classical x25519 KEM), the workload increases.
```